stutterhold/Source/PhaseVocodeur/PhaseVocodeur.cpp

```diff
@@ -48,7 +65,95 @@ PhaseVocodeur::PhaseVocodeur(int frame_size, int hop_size, int ola_size, int n_f
     init();
 }
 
-// ==============================================================================
+/*
+ ==============================================================================
+ Methods for initialization
+ ==============================================================================
+ */
+
+void PhaseVocodeur::init()
+{
+    /* Call all init subroutines */
+    init_ola();
+    init_window();
+    init_fft();
+}
+
+void PhaseVocodeur::init_ola()
+{
+    /* Initialize Overlap-Add (OLA) containers and rw positions */
+    
+    // assign number of frames based on overlap
+    num_ola_frames = ceil(static_cast<float>(ola_size) / static_cast<float>(hop_size));
+    
+    // allocate memory
+    ola_in = juce::AudioBuffer<float> (num_ola_frames, ola_size);
+    ola_out = juce::AudioBuffer<float> (num_ola_frames, ola_size);
+    
+    // clear
+    // @JulianArmandVanasse: Why? Cause residual junk maybe?
+    ola_in.clear(); ola_out.clear();
+    
+    // initialize rw positions
+    int pos = 0;
+    // @JulianArmandVanasse: Same Q^.
+    rw.clear();
+    for (int i = 0; i < num_ola_frames; i++)
+    {
+        rw.push_back(pos);
+        // @JulianArmandVanasse: Unclear on this, maybe knowing ola_size will help...
+        pos += (hop_size % ola_size);
+    }
+}
+
+void PhaseVocodeur::init_window()
+{
+    /* Initialize Hann window buffer. */
+    // Hann window is computed using the 'periodic' method.
+    
+    // allocate memory and clear
+    window = juce::AudioBuffer<float>(1, frame_size);
+    window.clear();
+    // fill using Hann function
+    
+    // @JulianArmandVannase: Why not `window[n]`?
+    auto w = window.getWritePointer(0);
+    
+    float N = static_cast<float>(frame_size);
+    for (int n = 0; n < frame_size; n++)
+    {
+        // @JulianArmandVanasse: Where are `ceil` `pow` and `sin` comming from? <math.h>?
+        // @JulianArmandVanasse: Why not a for loop with a float? Is that weird?
+        float fn = static_cast<float>(n);
+        
+        w[n] = pow(sin(M_PI * fn / N), 2.0f);
+    }
+}
+
+void PhaseVocodeur::init_fft()
+{
+    /* Initialize spectral containers and routines */
+    
+    // allocate space for input and output
+    fft_in = new kiss_fft_cpx[n_fft];
+    fft_out = new kiss_fft_cpx[n_fft];
+    
+    // initialize data storage
+    
+    // @JulianArmandVanasse: I'm scared...
+    memset(fft_in, 0, n_fft * sizeof(kiss_fft_cpx));
+    memset(fft_out, 0, n_fft * sizeof(kiss_fft_cpx));

```

Comment:
Is this a quick of the kiss_fft api, or is this something I should get used to seeing?